diem-node/src/lib.rs

```diff
@@ -276,8 +276,12 @@ pub fn setup_environment(node_config: &NodeConfig, logger: Option<Arc<Logger>>)
         node_config.storage.backup_service_address,
         Arc::clone(&diem_db),
     );
-
-    let genesis_waypoint = node_config.base.waypoint.genesis_waypoint();
+    let genesis_waypoint = node_config
```

Comment:
This is a good idea, but it breaks this approach FromStorage: https://github.com/diem/diem/pull/7505/files#diff-1dbb920b9fbe810ee8f189fe7aa8a360eb35febc32215a8261f5f23b44352b12R153

I think you'll want to hide the details of genesis_waypoint in the function and maybe not even make it pub

Reply:
@davidiw The link you mentioned in comment brings me back to the same file and doesn't have any other annotation. Can you please help me understand how it breaks WaypointConfig::FromStorage? (So that I can think of solving the issues)

I can create a function get_genesis_waypoint_config in BaseConfig, which can hide the implementation detail from its clients, but it will still have the same logic, i.e. if genesis_waypoint config is there, return that, or else return the waypoint config. But I think that is not what you are looking for. I think I don't understand the current FromStorage problem you pointed out.

Reply:
Do you mean in case genesis_waypoint is absent then waypoint will be returned and it will try to get diem_global_constants::GENESIS_WAYPOINT? In the absence of genesis_waypoint, do you just want the waypoint() to be called on waypoint config?

Reply:
well that is lame of me...

genesis_waypoint function actually does some additional resolution that this would break. So we want to backdoor this tweak in a backwards compatible fashion. So yes, in regard to your follow up question.

Reply:
Thank you. I will go ahead and implement the way I mentioned in second comment, but will it really keep it backwards compatible? Before this code, if there was a call to genesis_waypoint() with WaypointConfig::FromStorage, it used to fetch the diem_global_constants::GENESIS_WAYPOINT Waypoint. Now with the new implementation we are discussing, if there is not a genesis_waypoint defined in the BaseConfig and waypoint is WaypointConfig::FromStorage, it will fetch diem_global_constants::WAYPOINT waypoint. Is this what we intend? Is there some contract that if BaseConfig doesn't have a genesis_waypoint config, then diem_global_constants::GENESIS_WAYPOINT waypoint  doesn't exist  in the storage too?